### PR TITLE
added contributing to api

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
The content may be pretty obvious to many people but is wasn't to me when I started adding comments. It is an attempt to encourage others by giving them a starting point.

**Before merging can somebody review the following points:**

Is this the correct place for such a document?
The How to Contribute section is the way I found to do it is there a better way?
Have I got the terminology correct?
Is what I say about each entity correct and sensible?

Will edit or remove as decided

### CODE REVIEW

File: content/How_To/contribute/contribute_to_api.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+
+# Need for Contribution
+
+Many people have asked for the API documentation to be improved. This is a major task with lots of files needing comments added. Comments are very useful for future development and maintenance of the code but now they are even more useful since in the correct format they are can be read by “TYPEDOC” and produce the extra information API documentation for the classes, properties and methods used by Babylon.js. The core team has worked to make this happen as you can see by the new API documentation. Volunteers are needed to add comments so even if you only have time to do a couple of files then please do volunteer. 
+
+# How to Contribute
+
+1. Fork and clone [Babylon.js](https://github.com/BabylonJS/Babylon.js) from github;
+2. Edit a file from within the **src** folder by adding comments;
+3. Create a new folder **outside** the Babylon.js folder;
```
Comment (deltakosh) [inline content/How_To/contribute/contribute_to_api.md]: You don't need this one:)
We have a gulp task for that 
Just go to tools/gulp
Run npm  install
Run npm install -g gulp
Execute gulp typedoc-check

Comment (deltakosh) [reply content/How_To/contribute/contribute_to_api.md]: But your idea is also great as it help seeing the actual visual result

Comment (deltakosh) [reply content/How_To/contribute/contribute_to_api.md]: Mine has to run before a PR to make sure all comments are valid

File: content/How_To/contribute/contribute_to_api.md [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+## With Parameters
+
+A creates comment to describe the CONSTRUCTOR and use @param for each parameter. The first item after @param must be the parameter name then a further comment
+
+```javascript
+/**
+ * Creates a new Action
+ * @param triggerOptions the trigger, with or without parameters, for the action
+ * @param condition an optional determinant of action
```
Comment (deltakosh) [inline content/How_To/contribute/contribute_to_api.md:205]: Here it is a bit more tricky as you also have to duplicate the comment inside the parameters list for public members

File: content/How_To/contribute/contribute_to_api.md [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+/**
+ * Internal only - manager for action
+ * @ignore 
+ */
+public _actionManager: ActionManager;
+```
+
+## Private or Protected
+
+These will be ignored automatically when building the API documentation although plain comments are useful for development
```
Comment (deltakosh) [inline content/How_To/contribute/contribute_to_api.md]: We can say that this one is really optional:)

**Review by deltakosh — COMMENTED** [review]
I utterly appreciate your help John.
Thank you so much

File: content/How_To/contribute/contribute_to_api.md (hunk 1 above)
Comment (BabylonJSGuide) [reply content/How_To/contribute/contribute_to_api.md]: Did the install and gulp typedoc-check. Having run it it also does a typescript-compile and type-script-generate then a typedoc-validate. This makes changes to dist/preview release/babylon.d.ts which I can then either commit or discard. When I commit it and do gulp typedoc-check again I get a new set of differences to commit or discard. What do I do and what do I need to write in the instructions.

Comment (deltakosh) [reply content/How_To/contribute/contribute_to_api.md]: Correct it should not be committed actually :)

File: content/How_To/contribute/contribute_to_api.md (hunk 2 above)
Comment (BabylonJSGuide) [reply content/How_To/contribute/contribute_to_api.md:205]: Can and will emphasis this in next PR

File: content/How_To/contribute/contribute_to_api.md (hunk 3 above)
Comment (BabylonJSGuide) [reply content/How_To/contribute/contribute_to_api.md]: Is this what you mean

+## Private or Protected
+
+These will be ignored automatically when building the API documentation and comments are optional

Comment (deltakosh) [reply content/How_To/contribute/contribute_to_api.md]: Absolutely

File: content/How_To/contribute/contribute_to_api.md (hunk 1 above)
Comment (BabylonJSGuide) [reply content/How_To/contribute/contribute_to_api.md]: OK so I need to put in any instructions `discard changes suggested for babylon.d.ts`

Comment (deltakosh) [reply content/How_To/contribute/contribute_to_api.md]: Yes:)

**Comment by BabylonJSGuide** [comment]
Please review changes particularly under constructor

**Review by deltakosh — APPROVED** [review]
Perfect! Thanks a lot sir

**Pull request merged**